modules/websocket/wsl_client.cpp

```diff
@@ -231,6 +195,84 @@ int WSLClient::get_max_packet_size() const {
 }
 
 void WSLClient::poll() {
+	if (_status == Status::STATUS_RESOLVING) {
+		if (_resolving == IP::RESOLVER_INVALID_ID) {
+			_status = Status::STATUS_DISCONNECTED;
+			_on_error();
+			ERR_FAIL_MSG("Bug, invalid resolver ID");
+		}
+
+		IP::ResolverStatus rstatus = IP::get_singleton()->get_resolve_item_status(_resolving);
+		switch (rstatus) {
+			case IP::RESOLVER_STATUS_WAITING:
+				return; // Still resolving
+
+			case IP::RESOLVER_STATUS_DONE: {
+				_ip_candidates = IP::get_singleton()->get_resolve_item_addresses(_resolving);
+
+				if (_ip_candidates.size() == 0) {
+					_status = Status::STATUS_DISCONNECTED;
+					_on_error();
+					ERR_FAIL_MSG("Error, zero resolved hostnames");
+				}
+				_status = Status::STATUS_RESOLVED;
+			} break;
+			case IP::RESOLVER_STATUS_NONE:
+			case IP::RESOLVER_STATUS_ERROR: {
+				_status = Status::STATUS_DISCONNECTED;
+				_on_error();
+				ERR_FAIL_MSG("Error, couldn't resolve hostname");
+			} break;
+		}
+	}
+
+	if (_status == Status::STATUS_RESOLVED) {
+		String port = "";
+		if ((_port != 80 && !_use_ssl) || (_port != 443 && _use_ssl)) {
+			port = ":" + itos(_port);
+		}
+
+		Error err = ERR_BUG; // Should be at least one entry.
+		while (_ip_candidates.size() > 0) {
+			err = _tcp->connect_to_host(_ip_candidates.pop_front(), _port);
+			if (err == OK) {
+				break;
+			}
+		}
+		if (err != OK) {
+			_status = Status::STATUS_DISCONNECTED;
+			_tcp->disconnect_from_host();
+			_on_error();
+			return;
+		}
+		_connection = _tcp;
+
+		_key = WSLPeer::generate_key();
+		// TODO custom extra headers (allow overriding this too?)
```

Comment:
We can remove this TODO since we now support extra headers (was a leftover from the other PR I guess).